### ISSUE TITLE
Instance of status results showing up for wrong revision

### ISSUE THREAD

**Issue by imbstack** [taskcluster/taskcluster-github#151]
Not sure how widespread this is at the moment.

taskcluster/taskcluster-github#149 has statuses reported for 5cfa55d and  7c55062. However, the task groups linked in them are identical. We can see that different tasks were triggered for each revision [latter](https://github.taskcluster.net/v1/builds?organization=taskcluster&repository=taskcluster-github&sha=5cfa55d1f23d5402831640091d5f9f185c379429), [earlier](https://github.taskcluster.net/v1/builds?organization=taskcluster&repository=taskcluster-github&sha=7c55062203a4d65b13451cf3d5470988d97bd606).

**Comment by imbstack** [comment]
### Logs for 5cfa55d1f23d5402831640091d5f9f185c379429

```
    Feb 21 11:58:51 taskcluster-github app/web.1: Tue, 21 Feb 2017 19:58:51 GMT taskcluster-github:api:29a3b900-f870-11e6-895b-0843fbc0ed56 Received push event webhook payload. Processing... 
    Feb 21 11:58:51 taskcluster-github app/web.1: Tue, 21 Feb 2017 19:58:51 GMT taskcluster-github:api:29a3b900-f870-11e6-895b-0843fbc0ed56 Trying to authenticate as installation for push 
    Feb 21 11:58:51 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:58:51 GMT taskcluster-github:handlers:29a3b900-f870-11e6-895b-0843fbc0ed56 Received message. Starting processing... 
    Feb 21 11:58:51 taskcluster-github app/web.1: Tue, 21 Feb 2017 19:58:51 GMT taskcluster-github:api:29a3b900-f870-11e6-895b-0843fbc0ed56 Beginning publishing event message on pulse. 
    Feb 21 11:58:51 taskcluster-github app/web.1: Tue, 21 Feb 2017 19:58:51 GMT taskcluster-github:api:29a3b900-f870-11e6-895b-0843fbc0ed56 Finished Publishing event message on pulse. 
    Feb 21 11:58:52 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:58:51 GMT taskcluster-github:handlers:29a3b900-f870-11e6-895b-0843fbc0ed56 handling push webhook for: taskcluster/taskcluster-github@5cfa55d1f23d5402831640091d5f9f185c379429 
    Feb 21 11:58:52 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:58:51 GMT taskcluster-github:handlers:29a3b900-f870-11e6-895b-0843fbc0ed56 Trying to fetch the YML... 
    Feb 21 11:58:52 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:58:51 GMT taskcluster-github:handlers:29a3b900-f870-11e6-895b-0843fbc0ed56 Creating tasks. (taskGroupId: crIZbNU1Q9aD2Ew78a0gyQ) 
    Feb 21 11:58:53 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:58:52 GMT taskcluster-github:handlers:29a3b900-f870-11e6-895b-0843fbc0ed56 Trying to create status for taskcluster/taskcluster-github@5cfa55d1f23d5402831640091d5f9f185c379429 (pending) 
    Feb 21 12:00:55 taskcluster-github app/worker.1: Tue, 21 Feb 2017 20:00:54 GMT taskcluster-github:handlers:29a3b900-f870-11e6-895b-0843fbc0ed56 Handling state change for task-group crIZbNU1Q9aD2Ew78a0gyQ 
    Feb 21 12:00:55 taskcluster-github app/worker.1: Tue, 21 Feb 2017 20:00:55 GMT taskcluster-github:handlers:29a3b900-f870-11e6-895b-0843fbc0ed56 Authenticating as installation in status handler... 
    Feb 21 12:00:55 taskcluster-github app/worker.1: Tue, 21 Feb 2017 20:00:55 GMT taskcluster-github:handlers:29a3b900-f870-11e6-895b-0843fbc0ed56 Authorized as installation in status handler 
    Feb 21 12:00:55 taskcluster-github app/worker.1: Tue, 21 Feb 2017 20:00:55 GMT taskcluster-github:handlers:29a3b900-f870-11e6-895b-0843fbc0ed56 Attempting to update status for taskcluster/taskcluster-github@5cfa55d1f23d5402831640091d5f9f185c379429 (success) 
```

### Logs for 7c55062203a4d65b13451cf3d5470988d97bd606 (pull_request)

```
    Feb 21 11:42:15 taskcluster-github app/web.1: Tue, 21 Feb 2017 19:42:14 GMT taskcluster-github:api:d7fa2f00-f86d-11e6-884c-1cd119c72ce3 Received pull_request event webhook payload. Processing... 
    Feb 21 11:42:15 taskcluster-github app/web.1: Tue, 21 Feb 2017 19:42:14 GMT taskcluster-github:api:d7fa2f00-f86d-11e6-884c-1cd119c72ce3 Trying to authenticate as installation for pull_request 
    Feb 21 11:42:15 taskcluster-github app/web.1: Tue, 21 Feb 2017 19:42:15 GMT taskcluster-github:api:d7fa2f00-f86d-11e6-884c-1cd119c72ce3 Beginning publishing event message on pulse. 
    Feb 21 11:42:15 taskcluster-github app/web.1: Tue, 21 Feb 2017 19:42:15 GMT taskcluster-github:api:d7fa2f00-f86d-11e6-884c-1cd119c72ce3 Finished Publishing event message on pulse. 
    Feb 21 11:42:15 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:42:15 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-884c-1cd119c72ce3 Received message. Starting processing... 
    Feb 21 11:42:15 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:42:15 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-884c-1cd119c72ce3 handling pull_request.synchronize webhook for: taskcluster/taskcluster-github@7c55062203a4d65b13451cf3d5470988d97bd606 
    Feb 21 11:42:15 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:42:15 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-884c-1cd119c72ce3 Trying to fetch the YML... 
    Feb 21 11:42:15 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:42:15 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-884c-1cd119c72ce3 Checking pull request permission... 
    Feb 21 11:42:16 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:42:15 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-884c-1cd119c72ce3 Checking collaborator: imbstack is a collaborator on taskcluster/taskcluster-github: True! 
    Feb 21 11:42:16 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:42:15 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-884c-1cd119c72ce3 Creating tasks. (taskGroupId: EPjsINa7QKajF4B2JfSMMg) 
    Feb 21 11:42:17 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:42:16 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-884c-1cd119c72ce3 Trying to create status for taskcluster/taskcluster-github@7c55062203a4d65b13451cf3d5470988d97bd606 (pending) 
    Feb 21 11:44:35 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:44:35 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-884c-1cd119c72ce3 Handling state change for task-group EPjsINa7QKajF4B2JfSMMg 
    Feb 21 11:44:35 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:44:35 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-884c-1cd119c72ce3 Authenticating as installation in status handler... 
    Feb 21 11:44:35 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:44:35 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-884c-1cd119c72ce3 Authorized as installation in status handler 
    Feb 21 11:44:35 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:44:35 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-884c-1cd119c72ce3 Attempting to update status for taskcluster/taskcluster-github@7c55062203a4d65b13451cf3d5470988d97bd606 (success) 
```

### Logs for 7c55062203a4d65b13451cf3d5470988d97bd606 (push)

```
    Feb 21 11:42:15 taskcluster-github app/web.1: Tue, 21 Feb 2017 19:42:14 GMT taskcluster-github:api:d7fa2f00-f86d-11e6-8037-51b8acc8c9e7 Received push event webhook payload. Processing... 
    Feb 21 11:42:15 taskcluster-github app/web.1: Tue, 21 Feb 2017 19:42:14 GMT taskcluster-github:api:d7fa2f00-f86d-11e6-8037-51b8acc8c9e7 Trying to authenticate as installation for push 
    Feb 21 11:42:15 taskcluster-github app/web.1: Tue, 21 Feb 2017 19:42:15 GMT taskcluster-github:api:d7fa2f00-f86d-11e6-8037-51b8acc8c9e7 Beginning publishing event message on pulse. 
    Feb 21 11:42:15 taskcluster-github app/web.1: Tue, 21 Feb 2017 19:42:15 GMT taskcluster-github:api:d7fa2f00-f86d-11e6-8037-51b8acc8c9e7 Finished Publishing event message on pulse. 
    Feb 21 11:42:15 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:42:15 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-8037-51b8acc8c9e7 Received message. Starting processing... 
    Feb 21 11:42:15 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:42:15 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-8037-51b8acc8c9e7 handling push webhook for: taskcluster/taskcluster-github@7c55062203a4d65b13451cf3d5470988d97bd606 
    Feb 21 11:42:15 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:42:15 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-8037-51b8acc8c9e7 Trying to fetch the YML... 
    Feb 21 11:42:15 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:42:15 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-8037-51b8acc8c9e7 Creating tasks. (taskGroupId: VL1tMVSARMyjeHGfSUmwhA) 
    Feb 21 11:42:16 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:42:16 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-8037-51b8acc8c9e7 Trying to create status for taskcluster/taskcluster-github@7c55062203a4d65b13451cf3d5470988d97bd606 (pending) 
    Feb 21 11:44:35 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:44:34 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-8037-51b8acc8c9e7 Handling state change for task-group VL1tMVSARMyjeHGfSUmwhA 
    Feb 21 11:44:35 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:44:34 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-8037-51b8acc8c9e7 Authenticating as installation in status handler... 
    Feb 21 11:44:35 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:44:35 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-8037-51b8acc8c9e7 Authorized as installation in status handler 
    Feb 21 11:44:35 taskcluster-github app/worker.1: Tue, 21 Feb 2017 19:44:35 GMT taskcluster-github:handlers:d7fa2f00-f86d-11e6-8037-51b8acc8c9e7 Attempting to update status for taskcluster/taskcluster-github@7c55062203a4d65b13451cf3d5470988d97bd606 (success) 
```

**Comment by imbstack** [comment]
All statuses for [5cfa55d1f23d5402831640091d5f9f185c379429](https://api.github.com/repos/taskcluster/taskcluster-github/commits/5cfa55d1f23d5402831640091d5f9f185c379429/statuses).

All statuses for [7c55062203a4d65b13451cf3d5470988d97bd606](https://api.github.com/repos/taskcluster/taskcluster-github/commits/7c55062203a4d65b13451cf3d5470988d97bd606/statuses).

**Comment by imbstack** [comment]
Well... the Github api has correct results in it. This is either:

1. A Github UI bug.
2. Us misunderstanding what the Github UI is supposed to show.

Not sure what next-steps are here.

**Comment by imbstack** [comment]
Just for posterity's sake, I'll attach a screenshot showing the link of the pull_request status link on the PR. 

<img width="910" alt="screen shot 2017-02-21 at 2 39 41 pm" src="https://cloud.githubusercontent.com/assets/127521/23188538/e7bfbf4c-f843-11e6-8c84-de99fc19a24d.png">

As can be seen above, the task-group linked in that is triggered by an event from 7c55062203a4d65b13451cf3d5470988d97bd606 and not 5cfa55d1f23d5402831640091d5f9f185c379429. This is even though the api links above show the correct information.

**Comment by djmitche** [comment]
![page-shot-2017-2-21 report yaml error to users by imbstack pull request 149 taskcluster taskcluster-github](https://cloud.githubusercontent.com/assets/28673/23188628/6bd341d2-f85d-11e6-9174-1fff53028813.png)

I don't even see 5cfa55d.  The two "X" statuses that appear have four different taskgroups linked.  I suspect this is a UI issue..

**Comment by imbstack** [comment]
Ok, turns out this isn't a bug in github, but rather what I would consider confusing github ui. As explained by github support (who were really awesome about this):

> Statuses can be created only for commits, not for pull requests. The statuses shown in the pull request's merge box are always taken from the latest commit in the pull request's head branch -- the latest per-context statuses are shown (i.e. if there are multiple statuses for the same context, the latest status with that context is shown).
>
> Now, here's the important bit. When the pull request is merged -- we remember which statues were created for the latest commit in the pull request's head branch up until that point in time. After the pull request is merged -- you can click the "view details" button to see the statuses for the pull request. And what you will see are statuses as they were at the moment when the pull request was merged. However, statuses for a specific commit will still show the latest per-context statuses at the moment when you click the x or checkmark next to the commit SHA.
>
> This difference is relevant because it's possible to create statuses on commits even after a pull request was merged. Imagine if there was a status in pending state for context foo at the moment when the pull request was merged. After it was merged, a new status with context foo but with a success state was created for the same commit. The UI for the commit's statuses will show the latest state -- context foo has a success status. However, the merge box for the pull request (after you click "view details") will tell you that at the moment when the pull request was merged -- the foo context had only a pending state status.
>
> And that's what happened in your case. Statuses from commit 7c55062 were taken for the pull request because that was the latest commit in the pull request at the time. And only those statuses that were created up until that moment are shown when you click the "view details" button. But because more statuses were created after the pull request was merged, the UI for commit 7c55062 shows something different.
>
> You can check this by looking at the dates when the statuses were created and considering only those that were created before the date that the pull request was merged on.
